sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -659,12 +677,70 @@ def crop(self, value):
             raise TileSourceError(msg)
         self._crop = (x, y, w, h)
 
+    def _generateDownsampledLevels(self, resample_method):
+        self._checkEditable()
+        current_arrays = dict(self._zarr.arrays())
+        if '0' not in current_arrays:
+            msg = 'No root data found, cannot generate lower resolution levels.'
+            raise TileSourceError(msg)
+        if 'x' not in self._axes or 'y' not in self._axes:
+            msg = 'Data must have an X axis and Y axis to generate lower resolution levels.'
+            raise TileSourceError(msg)
+
+        metadata = self.getMetadata()
+        tile_size = dict(width=4096, height=4096)
```

Comment:
For efficiency reasons, we might want this to be 4096+(overlap size).  More on overlap below.

Reply:
Changed in [9a74c0a](https://github.com/girder/large_image/pull/1490/commits/9a74c0ac85f7c6394c25d0a29e395f0c371f337f)